distribution/packages/src/common/scripts/preinst

```diff
@@ -9,6 +9,18 @@
 #       $1=1       : indicates an new install
 #       $1=2       : indicates an upgrade
 
+# Check for these at preinst time due to failures in postinst if they do not exist
+if [ -x "$JAVA_HOME/bin/java" ]; then
+    JAVA="$JAVA_HOME/bin/java"
+else
+    JAVA=`which java`
+fi
+
+if [ ! -x "$JAVA" ]; then
```

Comment:
Does this ever execute in a failing case? I think that these scripts are always executed with `set +e` or does it differ by packaging system? If we get here it's because `$JAVA_HOME/bin/java` exists which we set `JAVA` to, or `which java` failed and we should have exited?

Reply:
yea i copied this from https://github.com/elastic/elasticsearch/blob/master/distribution/packages/src/deb/init.d/elasticsearch#L83-L94 , and i think the logic is wrong. Ill fix up.

Reply:
Changed the bash test check to see if `$JAVA` is set.